xdsl/ir/core.py

```diff
@@ -61,6 +61,14 @@ def attributes(self) -> Iterator[type[Attribute]]:
     def name(self) -> str:
         return self._name
 
+    @staticmethod
+    def split_name(name: str) -> tuple[str, str]:
+        x = name.split(".", 1)
+        if len(x) == 2:
+            return (x[0], x[1])
+        else:
+            return (name, "")
```

Comment:
I don't think we want the special logic, we might as well assert that there are two components, it's user error to provide a string that is neither an operation nor an attribute name.
```suggestion
        d, rest = name.split(".", 1)
        return (d, rest)
```

Reply:
That makes sense, is it worth throwing a custom exception or just let the IndexError get thrown?

Reply:
I guess it would be a little nicer to have a custom exception, I'll reword my suggestion